src/main/java/org/spongepowered/common/data/util/ValueProcessorDelegate.java

```diff
@@ -109,11 +110,17 @@ public DataTransactionResult offerToStore(ValueContainer<?> container, E value)
         }
         for (ValueProcessor<E, V> processor : this.processors) {
             if (processor.supports(container)) {
-                final Optional<V> optional = processor.getApiValueFromContainer(container);
-                if (optional.isPresent()) {
-                    V mutable = optional.get();
-                    ((Value<E>) mutable).set(value);
-                    return DataTransactionResult.failResult(((Value<E>) mutable).asImmutable());
+                final Optional<V> currentValueOptional = processor.getApiValueFromContainer(container);
+                if (currentValueOptional.isPresent()) {
+                    V currentValue = currentValueOptional.get();
+                    Value<E> rejectedValue;
```

Comment:
Leaving aside the argument of what `getApiValueFromContainer` should be returning for the moment, if we go ahead with this way, this should be an `ImmutableValue`. If you have an `ImmutableValue`, it'll will be converted to a `Value` then back to an `ImmutableValue` again for no particular reason. Minimise conversions please.

Reply:
I hope I understand you correctly. I made some changes in the next commit.